src/resources/views/crud/fields/address_algolia.blade.php

```diff
@@ -9,7 +9,7 @@
         'field' => $field['name'],
         'full' => $field['store_as_json'],
     ];
-    $field['value'] = old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? '';
+    $field['value'] = oldValueDefaultOrFallback($field, '');
```

Comment:
Note to self - I think this would be a much smaller change if we don't include the field the helper (which we probably shouldn't anyway, since helpers are supposed to be as general as possible.

If we had this:

```php
    /**
     * Laravel's ConvertEmptyStringsToNull middleware makes it impossible know when a
     * form input was submitted as "null", empty string or just didn't exist at all.
     * This helper is here as an alternative to old() that returns a fallback
     * value in case the input wasn't submitted at all.
     *
     * The differences are that:
     * - it also works for nested inputs using array notation - testimonial[name];
     * - it also works for nested inputs using dot notation - testionial.name;
     *
     * @param  string  $input_name
     * @param  mixed  $fallback_value
     * @return mixed
     */
    function old_input_value($input_name, $fallback_value = null)
    {
        $input_name = square_brackets_to_dots($input_name);
        $old_inputs = session()->getOldInput();

        return \Arr::get($old_inputs, $input_name) ?? $fallback_value;
    }
```

We could do this:

```suggestion
    $field['value'] = old_input_value($field['name'], $field['value'] ?? $field['default'] ?? '');
```

Reply:
@tabacitu please note the fix for the previous mentioned problem includes checking if one key exists in the array of field. 

You solution "pre-parses" that value so we wouldn't know if we received a field value or a default or the fallback.

Reply:
> one key exists in the array of field.

I don't understand, sorry. Please rephrase?

Reply:
Yeah no... I still think this is the best path forward... with the old_input_value() I've suggested above. @pxpm please let me know why you think this is a bad idea... save me time before I dig in and do it 🙏